src/memefactory/server/graphql_resolvers.cljs

```diff
@@ -1,16 +1,687 @@
+                                    :challenge/commit-period-end :challenge/commit-period-end
+                                    :challenge/reveal-period-end :challenge/votes-for :challenge/votes-against]}]
+
+  (cond
+    (and (< now challenge-period-end) (not challenger)) :reg-entry.status/challenge-period
+    (< now commit-period-end)                           :reg-entry.status/commit-period
+    (< now reveal-period-end)                           :reg-entry.status/reveal-period
+    (or (< votes-against votes-for)
+        (< challenge-period-end now))                   :reg-entry.status/whitelisted
+    :else                                               :reg-entry.status/blacklisted))
+
+(defn meme-status-sql-clause [now]
+  (sql/call ;; TODO: can we remove aliases here?
+   :case
+   [:and
+    [:< now :re.reg-entry/challenge-period-end]
+    [:= :re.challenge/challenger nil]]                        (enum :reg-entry.status/challenge-period)
+   [:< now :re.challenge/commit-period-end]                   (enum :reg-entry.status/commit-period)
+   [:< now :re.challenge/reveal-period-end]                   (enum :reg-entry.status/reveal-period)
+   [:or
+    [:< :re.challenge/votes-against :re.challenge/votes-for]
+    [:< :re.reg-entry/challenge-period-end now]]              (enum :reg-entry.status/whitelisted)
+   :else                                                      (enum :reg-entry.status/blacklisted)))
+
+(defn search-memes-query-resolver [_ {:keys [:statuses :order-by :order-dir :owner :creator :curator :first :after] :as args}]
+  (log/debug "search-memes-query-resolver" args)
+  (let [statuses-set (when statuses (set statuses))
+        now (last-block-timestamp)
+        page-start-idx (when after (js/parseInt after))
+        page-size first
+        order-by-clause (when order-by
+                          {:order-by [[(get {:memes.order-by/reveal-period-end    :re.challenge/reveal-period-end
+                                             :memes.order-by/commited-period-end  :re.challenge/commit-period-end
+                                             :memes.order-by/challenge-period-end :re.reg-entry/challenge-period-end
+                                             :memes.order-by/total-trade-volume   :m.meme/total-trade-volume
+                                             :memes.order-by/created-on           :re.reg-entry/created-on
+                                             :memes.order-by/number               :m.meme/number
+                                             :memes.order-by/total-minted         :m.meme/total-minted}
+                                            ;; TODO: move this transformation to district-server-graphql
+                                            (graphql-utils/gql-name->kw order-by))
+                                       (or (keyword order-dir) :asc)]]})
+        where-clause (when (or curator creator owner statuses-set)
```

Comment:
You can rather use `honeysql.helpers/merge-where` and `honeysql.helpers/merge-order-by`